src/components/Search.js

```diff
@@ -1,17 +1,45 @@
 import React from 'react';
+import axios from 'axios';
 
 class Search extends React.Component {
+  state = {
+    query: ''
+  }
+
+  _onChange = e => {
+    const { name, value } = e.target;
+    this.setState({
+      [name]: value
+    });
+  }
+
+  _onSubmit = (e) => {
+    e.preventDefault();
+    const { query } = this.state;
+    const { onError, onSearch } = this.props;
+
+    axios.get(`http://api.giphy.com/v1/gifs/search?q=${query}&api_key=q7kTB0TwzgVggIoYvFUcT97sfeuEIL53`)
+     .then(response => {
+       onSearch(response.data.data);
+     })
+     .catch(error => {
+       onError(error.message);
+     });
+   }
+
   render() {
     return (
       // Search bar which communicates with giphy API
       <div className="wrap">
         <div className="search">
-          <input type="text" className="searchTerm" placeholder="What are you looking for?" />
-            <button onClick={this.props.userAction}  type="submit" className="searchButton">
+          <form onSubmit={this._onSubmit}>
+            <input type="text" name='query' onChange={this._onChange} className="searchTerm" placeholder="What are you looking for?" />
+            <button type="submit" className="searchButton">
               <i className="fa fa-search"></i>
             </button>
-          </div>
+          </form>
         </div>
+      </div>
     );
```

Comment:
To get the expected searching when a user presses enter, the input needs to be moved into a `<form>` tag with `onSubmit` for the form being used instead on an `onClick` on a button.